src/compiler/types.ts

```diff
@@ -5936,6 +5936,7 @@ export interface Symbol {
     /** @internal */ constEnumOnlyModule: boolean | undefined; // True if module contains only const enums or other modules with only const enums
     /** @internal */ isReferenced?: SymbolFlags; // True if the symbol is referenced elsewhere. Keeps track of the meaning of a reference in case a symbol is both a type parameter and parameter.
     /** @internal */ lastAssignmentPos?: number; // Source position of last node that assigns value to symbol
+    /** @internal */ isDefinitelyAssigned?: boolean; // True if the symbol is assigned definitely

```

Comment:
I think there's a different use of terminology here. In most interpretations, definitely-assigned implies that a variable is *always* assigned to by a specific point.

Here, it feels like the meaning is "some assignment definitely exists".

Reply:
🤔  what about `hasDefiniteAssignment`